src/parser/xmlHandler.js

```diff
@@ -823,7 +823,13 @@ function parseValue(text, descriptor) {
   var value = text;
   var jsType = descriptor && descriptor.jsType;
   if (jsType === Date) {
-    value = new Date(text);
+    var dateText = text;
+    // Checks for xs:date with tz 
+    // (examples: 2019-03-26Z or 2019-03-26-06:00)
+    if(dateText.endsWith('Z') || dateText.length === 16){
+      dateText = text.substr(0, 10);
+    }
+    value = new Date(dateText);
```

Comment:
In my test:

```
> new Date('2019-03-26Z')
2019-03-26T00:00:00.000Z
> new Date('2019-03-26')
2019-03-26T00:00:00.000Z
> new Date('2019-03-26Z06:00')
2019-03-26T06:00:00.000Z
> new Date('2019-03-26-06:00')
Invalid Date
```

Which date format do you try to support? Is it a valid xsd:date or xsd:dateTime?

Reply:
Valid date formats with TZ for xsd:date are

2002-09-24Z
2002-09-24-06:00
2002-09-24+06:00

Taken from: https://www.w3schools.com/xml/schema_dtypes_date.asp

xsd:dateTime has no problem because it includes time and parses as is.

Reply:
I don't think we can just drop `-06:00` or `+06:00` to support the following xml dates:

- 2002-09-24-06:00
- 2002-09-24+06:00

Please also add some comments and tests to cover the cases.

Reply:
The thing is that the Date object cannot store an arbitrary timezone, it just takes the time as is and puts the local machine timezone. For representation purposes, the TZ cannot be stored unless we create a "custom" Date object, but I think that could break functionality.

My best suggestion is to drop the TZ information, because in this case there is no time to calculate an offset.

I need some directions to where write tests to this, I tried with a dummy request-response-sample but it doesn't execute my code.

Reply:
As far as I understand, `Date` objects use a Unix Time Stamp, an integer value that is the number of milliseconds since 1 January 1970 UTC. So it's agonistic of time zone.

Reply:
Yeah, that's basically it, there's no way to store the TZ info, that's why I set it to discard it when there's only date without time.

Do I have to do anything else for this to be merged or just wait for @b-admike 's review?, We need this to be able to work with a soap service that returns dates with tz info.